api/v1alpha1/warehouse_types.go

```diff
@@ -55,13 +55,30 @@ type WarehouseSpec struct {
 	// empty, the defaulting webhook will set the value of this field to the value
 	// of the shard label.
 	Shard string `json:"shard,omitempty" protobuf:"bytes,2,opt,name=shard"`
+	// FreightCreation describes how Freight is created by this Warehouse.
+	// This field is optional. When left unspecified, the field is implicitly
+	// treated as if its value were "Automatic".
+	//
+	// +kubebuilder:default=Automatic
+	FreightCreation FreightCreation `json:"freightCreation" protobuf:"bytes,3,opt,name=freightCreation"`
```

Comment:
I am not too keen on how this field has been named. Happy to take suggestions.

Reply:
Only thing I can think of that might be a very slight improvement is `FreightCreationPolicy`.